packages/legacy/src/Plans/components/Wizard/GeneralForm.tsx

```diff
@@ -50,10 +52,21 @@ export const GeneralForm: React.FunctionComponent<IGeneralFormProps> = ({
   const inventoryProvidersQuery = useInventoryProvidersQuery();
   const clusterProvidersQuery = useClusterProvidersQuery(namespace);
   const namespacesQuery = useNamespacesQuery(form.values.targetProvider);
+  const secretsQuery = useSecretsForProvidersQuery(clusterProvidersQuery, namespace);
+
   const targetNsFoundInQueriesNs = !!namespacesQuery.data?.find(
     (namespace) => form.values.targetNamespace === namespace.name
   );
 
+  const [sourceClusterProvider, setSourceClusterProvider] = React.useState<IProviderObject>(); // TODO: Use a default for edit form
+  const isSourceProviderInsecure = React.useMemo(() => {
+    const secretName = sourceClusterProvider?.spec.secret.name;
```

Comment:
according to `V1beta1ProviderSpec` every field is optional so we should probably continue chaining `?.`